pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -54,21 +57,33 @@ type LaunchTemplateProvider struct {
 	amiProvider           *AMIProvider
 	securityGroupProvider *SecurityGroupProvider
 	cache                 *cache.Cache
+	clusterDNSIP          net.IP
 }
 
-func NewLaunchTemplateProvider(ctx context.Context, ec2api ec2iface.EC2API, amiProvider *AMIProvider, securityGroupProvider *SecurityGroupProvider) *LaunchTemplateProvider {
+func NewLaunchTemplateProvider(ctx context.Context, ec2api ec2iface.EC2API, clientSet *kubernetes.Clientset, amiProvider *AMIProvider, securityGroupProvider *SecurityGroupProvider) *LaunchTemplateProvider {
 	l := &LaunchTemplateProvider{
 		ec2api:                ec2api,
 		logger:                logging.FromContext(ctx).Named("launchtemplate"),
 		amiProvider:           amiProvider,
 		securityGroupProvider: securityGroupProvider,
 		cache:                 cache.New(CacheTTL, CacheCleanupInterval),
+		clusterDNSIP:          discoverClusterDNS(ctx, clientSet),
 	}
 	l.cache.OnEvicted(l.onCacheEvicted)
 	l.hydrateCache(ctx)
 	return l
 }
 
+func discoverClusterDNS(ctx context.Context, clientSet *kubernetes.Clientset) net.IP {
+	dnsService, err := clientSet.CoreV1().Services("default").Get(ctx, "kubernetes", metav1.GetOptions{})
+	if err != nil {
+		logging.FromContext(ctx).Debugf("Unable to discover clusterDNS, %v", err)
+		return nil
+	}
+	logging.FromContext(ctx).Debugf("Discovered clusterDNS, %s", dnsService.Spec.ClusterIP)
+	return net.ParseIP(dnsService.Spec.ClusterIP)
```

Comment:
This wouldn't support dual stack right? https://kubernetes.io/docs/concepts/services-networking/dual-stack/

Reply:
In dual stack mode we can get IPv6 address from the field if the user prefers the address family. So it seems to be able to support dual stack.